FirebaseRemoteConfig/Sources/RCNConfigRealtime.m

```diff
@@ -432,37 +432,40 @@ - (void)fetchLatestConfig:(NSInteger)remainingAttempts targetVersion:(NSInteger)
       }
     } else {
       [strongSelf->_configFetch
-          fetchConfigWithExpirationDuration:0
-                          completionHandler:^(FIRRemoteConfigFetchStatus status, NSError *error) {
-                            if (error != nil) {
-                              FIRLogError(kFIRLoggerRemoteConfig, @"I-RCN000010",
-                                          @"Failed to retrive config due to fetch error. Error: %@",
-                                          error);
-                              [self propogateErrors:error];
-                            } else {
-                              if (status == FIRRemoteConfigFetchStatusSuccess) {
-                                if ([strongSelf->_configFetch.templateVersionNumber integerValue] >=
-                                    targetVersion) {
-                                  for (RCNConfigUpdateCompletion listener in strongSelf
-                                           ->_listeners) {
-                                    listener(nil);
-                                  }
-                                } else {
-                                  FIRLogDebug(kFIRLoggerRemoteConfig, @"I-RCN000016",
-                                              @"Fetched config's template version is outdated, "
-                                              @"re-fetching");
-                                  [strongSelf autoFetch:remainingAttempts - 1
-                                          targetVersion:targetVersion];
-                                }
-                              } else {
-                                FIRLogDebug(
-                                    kFIRLoggerRemoteConfig, @"I-RCN000016",
-                                    @"Fetched config's template version is outdated, re-fetching");
-                                [strongSelf autoFetch:remainingAttempts - 1
-                                        targetVersion:targetVersion];
-                              }
-                            }
-                          }];
+          realtimeFetchConfigWithNoExpirationDuration:gFetchAttempts - remainingAttempts + 1
```

Comment:
nit: you could just decrement `remainingAttempts` at the beginning of the method, then you don't need the "+1" here and you can directly pass `remainingAttempts` to the next autofetch call